core/src/main/scala/org/apache/spark/SparkEnv.scala

```diff
@@ -302,7 +301,26 @@ object SparkEnv extends Logging {
     val broadcastManager = new BroadcastManager(isDriver, conf, securityManager)
 
     val mapOutputTracker = if (isDriver) {
-      new MapOutputTrackerMaster(conf, broadcastManager, isLocal)
+      val master = conf.get("spark.master")
+      val shuffleProvider = conf.get(SHUFFLE_SERVICE_PROVIDER_CLASS)
+        .map(clazz => Utils.loadExtensions(
+          classOf[ShuffleServiceAddressProviderFactory],
+          Seq(clazz), conf)).getOrElse(Seq())
+      val serviceLoaders = shuffleProvider
+        .filter(_.canCreate(conf.get("spark.master")))
+      if (serviceLoaders.size > 1) {
```

Comment:
can this be more than one? you're only ever specifying a single class in the conf? but I could be misunderstanding this